spec/mailers/subscription_mailer_spec.rb

```diff
@@ -240,15 +240,22 @@
         allow(summary).to receive(:order_count) { 37 }
         allow(summary).to receive(:issue_count) { 0 }
         allow(summary).to receive(:issues) { {} }
-        SubscriptionMailer.placement_summary_email(summary).deliver_now
       end
 
       it "sends the email, which notifies the enterprise that all orders were successfully processed" do
+        SubscriptionMailer.placement_summary_email(summary).deliver_now
+
         expect(body).to include I18n.t("#{scope}.placement_summary_email.intro", shop: shop.name)
         expect(body).to include I18n.t("#{scope}.summary_overview.total", count: 37)
         expect(body).to include I18n.t("#{scope}.summary_overview.success_all")
         expect(body).to_not include I18n.t("#{scope}.summary_overview.issues")
       end
+
+      it "renders the shop's logo" do
+        shop.update!(logo: fixture_file_upload("logo.png", "image/png"))
+        SubscriptionMailer.placement_summary_email(summary).deliver_now
+        expect(SubscriptionMailer.deliveries.last.body).to include "logo.png"
+      end
```

Comment:
Hey @mkllnk ,
I've ran this spec file locally, these changes above: 
it passes, even without changing `logo_variant` -> `logo_url` on the file `app/views/subscription_mailer/placement_summary_email.html.haml`, which, if I understand correctly, is actually fixing the bug.

I don't have an improvement suggestion for the spec right now - just noting that we might be missing the coverage on the code which triggered the initial issue.

Reply:
That's odd. I just checked out master, changed the method and ran the spec:
```
$ git diff
diff --git a/app/views/subscription_mailer/placement_summary_email.html.haml b/app/views/subscription_mailer/placement_summary_email.html.haml
index 4cdefdffe..3c8a21bac 100644
--- a/app/views/subscription_mailer/placement_summary_email.html.haml
+++ b/app/views/subscription_mailer/placement_summary_email.html.haml
@@ -12,7 +12,7 @@
         %tr
           %td{:align => "right"}
             - if @shop.logo.variable?
-              = image_tag @shop.logo_url(:medium), class: "float-right"
+              = image_tag @shop.logo_variant(:medium), class: "float-right"
       %span.clear
 
 = render 'summary_overview', summary: @summary


$ bundle exec rspec spec/mailers/subscription_mailer_spec.rb
[TEST PROF INFO] Spring detected
DEPRECATION WARNING: Initialization autoloaded the constants ReportsHelper, DateTimeStringValidator, and IntegerArrayValidator.

Being able to do this is deprecated. Autoloading during initialization is going
to be an error condition in future versions of Rails.

Reloading does not reboot the application, and therefore code executed during
initialization does not run again. So, if you reload ReportsHelper, for example,
the expected changes won't be reflected in that stale Module object.

These autoloaded constants have been unloaded.

In order to autoload safely at boot time, please wrap your code in a reloader
callback this way:

    Rails.application.reloader.to_prepare do
      # Autoload classes and modules needed at boot time here.
    end

That block runs when the application boots, and every time there is a reload.
For historical reasons, it may run twice, so it has to be idempotent.

Check the "Autoloading and Reloading Constants" guide to learn more about how
Rails autoloads and reloads.
 (called from <main> at /home/maikel/code/openfoodnetwork/config/environment.rb:5)
Running via Spring preloader in process 27978
Knapsack adapter is off!
.................FFFFF....

Failures:

  1) SubscriptionMailer order placement summary when no issues were encountered while processing subscriptions sends the email, which notifies the enterprise that all orders were successfully processed
     Failure/Error: = image_tag @shop.logo_variant(:medium), class: "float-right"
     
     ActionView::Template::Error:
       ...
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `logo_variant' for #<Enterprise id: 598, name: "Enterprise 47", description: "enterprise", ...     #   Did you mean?  logo_attachment_id
     #   ./app/views/subscription_mailer/placement_summary_email.html.haml:15:in `_app_views_subscription_mailer_placement_summary_email_html_haml__3981160713829731655_162960'

Finished in 29.83 seconds (files took 0.94576 seconds to load)
26 examples, 5 failures

Failed examples:

rspec ./spec/mailers/subscription_mailer_spec.rb:245 # SubscriptionMailer order placement summary when no issues were encountered while processing subscriptions sends the email, which notifies the enterprise that all orders were successfully processed
rspec ./spec/mailers/subscription_mailer_spec.rb:254 # SubscriptionMailer order placement summary when no issues were encountered while processing subscriptions renders the shop's logo
rspec ./spec/mailers/subscription_mailer_spec.rb:275 # SubscriptionMailer order placement summary when some issues were encountered while processing subscriptions when no unrecorded issues are present sends the email, which notifies the enterprise that some issues were encountered
rspec ./spec/mailers/subscription_mailer_spec.rb:302 # SubscriptionMailer order placement summary when some issues were encountered while processing subscriptions when some undocumented orders are present sends the email, which notifies the enterprise that some issues were encountered
rspec ./spec/mailers/subscription_mailer_spec.rb:330 # SubscriptionMailer order placement summary when no subscriptions were processed successfully sends the email, which notifies the enterprise that some issues were encountered
```